internal/dcache/cluster_manager/cluster_manager.go

```diff
@@ -1210,51 +1175,62 @@ func (cmi *ClusterManager) updateRVList(existingRVMap map[string]dcache.RawVolum
 	now := uint64(time.Now().Unix())
 	hbExpiry := now - uint64(hbTillNodeDown*hbSeconds)
 
-	// (1) Update RVs present in existingRVMap and which have changed State or AvailableSpace.
-	for rvName, rvInClusterMap := range existingRVMap {
-
-		if rvHb, found := rVsByRvIdFromHB[rvInClusterMap.RvId]; found {
-			lastHB := rvLastHB[rvHb.RvId]
+	// (1.a) If it is only myRV, verify if it is not already present in existingRVMap.
+	if onlyMyRVs {
+		for rvName, rvInClusterMap := range existingRVMap {
+			if _, found := rVsByRvIdFromHB[rvInClusterMap.RvId]; found {
+				log.Warn("ClusterManager::updateRVList: RV %s is already present in clusterMap, skipping it",
+					rvName)
+				return false, nil
+			}
```

Comment:
since there could be more than one RV, we cannot just return if we find one.
It'll be simpler to let onlyMyRvs case also run through the following loops and only skip the "RV present in existingRVMap, but missing from rVsByRvIdFromHB" case.